AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/Resolvers/Authorization/AuthorizationState+Resolver.swift

```diff
@@ -26,32 +26,88 @@ extension AuthorizationState {
                     return .from(oldState)
                 }
                 return resolveNotConfigured(byApplying: authZEvent)
+
             case .configured:
-                guard case .fetchAuthSession(let cognitoCredentials) = isAuthorizationEvent(event)?.eventType else {
-                    return .from(oldState)
+
+                if let authenEvent = event as? AuthenticationEvent,
+                   case .signInRequested = authenEvent.eventType {
+                    return .init(newState: .signingIn)
                 }
-                return resolveFetchAuthSessionEvent(storedCredentials: cognitoCredentials)
-            case .sessionEstablished, .error:
-                guard case .configure = isAuthorizationEvent(event)?.eventType else {
-                    return .from(oldState)
+
+                if case .fetchUnAuthSession = isAuthorizationEvent(event)?.eventType {
+                    let action = InitializeFetchUnAuthSession()
+                    return .init(newState: .fetchingUnAuthSession(.notStarted), actions: [action])
                 }
-                return .init(newState: .configured)
-            case .fetchingAuthSession(let fetchAuthSessionState):
-                let fetchAuthSessionStateResolver = FetchAuthSessionState.Resolver()
-                let fetchAuthSessionResolution = fetchAuthSessionStateResolver.resolve(
-                    oldState: fetchAuthSessionState, byApplying: event)
-                guard case let .fetchedAuthSession(sessionData) = isAuthorizationEvent(event)?.eventType else {
-                    let authorizationState = AuthorizationState.fetchingAuthSession(fetchAuthSessionResolution.newState)
-                    return .init(newState: authorizationState, actions: fetchAuthSessionResolution.actions)
+
+                return .from(oldState)
+
+
+            case .sessionEstablished:
+                if let authenEvent = event as? AuthenticationEvent,
+                   case .signInRequested = authenEvent.eventType {
+                    return .from(.signingIn)
                 }
-                // Move the Authorization back to configured
-                let action = InitializeAuthorizationConfiguration()
-                let resolution = StateResolution(
-                    newState: AuthorizationState.sessionEstablished(sessionData),
-                    actions: [action]
-                )
-                return resolution
+                return .from(oldState)
+
+            case .signingIn:
+                if let authEvent = event as? AuthenticationEvent {
+                    switch authEvent.eventType {
+                    case .signInCompleted(let event):
+                        let action = InitializeFetchUnAuthSession()
```

Comment:
Shouldn't this action be `InitializeFetchAuthSession` rather than `UnAuth`?

Reply:
Yes, changed to InitializeFetchAuthSessionWithUserPool